app/models/organisation.rb

```diff
@@ -8,7 +8,10 @@ class Organisation < ActiveRecord::Base
   has_many :sub_organisations, -> { order :id }, class_name: "Organisation", foreign_key: "parent_organisation_id"
   belongs_to :parent_organisation, class_name: "Organisation"
 
+  acts_as_url :name, url_attribute: :slug, only_when_blank: true
+
   validates :slug, :name, presence: true
+  validates :slug, uniqueness: true
```

Comment:
This needs a migration to add a DB level unique constraint too. Without it it does "if SELECT fails, INSERT" which in a race (such as perhaps a double click from a user on the submit form) can end up with two records.